File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    void reverse(int[] nums, int s, int end){
        while(s<end){
            int temp= nums[s];
            nums[s]= nums[end];
            nums[end]= temp;
            s++;
            end--;

        }
    }
    public void rotate(int[] nums, int k) {
        int n= nums.length;
        k= k%n;
        int s= 0;
        int end=n-1;
        reverse(nums,0,n-1);
        reverse(nums, 0, k-1);
        reverse(nums, k, n-1);
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
void reverse(vector<int>&nums, int s, int end){
    while(s<end){
        swap(nums[s],nums[end]);
        s++;
        end--;
    }
}
    void rotate(vector<int>& nums, int k) {
       int n= nums.size();
       k= k%n;
       reverse(nums, 0,n-1);
       reverse(nums,0,k-1);
       reverse(nums, k, n-1);
       
    }
};